android/app/src/main/java/app/organicmaps/location/TrackRecordingService.java

```diff
@@ -158,20 +159,18 @@ public int onStartCommand(@NonNull Intent intent, int flags, int startId)
       return START_NOT_STICKY;
     }
 
-    Logger.i(TAG, "Starting foreground service");
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S)
+    Logger.i(TAG, "Starting Track Recording Foreground service");
+
+    try
     {
-      try
-      {
-        startForeground(TrackRecordingService.TRACK_REC_NOTIFICATION_ID, getNotificationBuilder(this).build());
-      } catch (ForegroundServiceStartNotAllowedException e)
-      {
-        Logger.e(TAG, "Oops! ForegroundService is not allowed", e);
-      }
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q)
+        ServiceCompat.startForeground(this, TrackRecordingService.TRACK_REC_NOTIFICATION_ID, getNotificationBuilder(this).build(), ServiceInfo.FOREGROUND_SERVICE_TYPE_MANIFEST);
+      else
+        ServiceCompat.startForeground(this, TrackRecordingService.TRACK_REC_NOTIFICATION_ID, getNotificationBuilder(this).build(), 0);
     }
-    else
+    catch (Exception e)
     {
-      startForeground(TrackRecordingService.TRACK_REC_NOTIFICATION_ID, getNotificationBuilder(this).build());
+      Logger.e(TAG, "Unable to start Track Recording Foreground Service", e);
```

Comment:
I don't like silent failure approach, it would be hard to debug later. If permissions are checked properly before launching the service, then there should be no any exceptions. And if there will be any exception, it would be important for us to know about it ASAP to fix properly instead of leaving a user with a non-working service.